reactos/drivers/storage/storahci/storahci.c

```diff
@@ -623,29 +669,38 @@ ULONG DriverEntry(
  * @return
  * TRUE in case AHCI Controller RESTARTED successfully. i.e GHC.HR == 0
  */
-BOOLEAN AhciAdapterReset(
-  __in      PAHCI_ADAPTER_EXTENSION             adapterExtension
-)
+BOOLEAN
+AhciAdapterReset (
+    __in      PAHCI_ADAPTER_EXTENSION             adapterExtension
+    )
 {
     ULONG ghc, ticks;
     PAHCI_MEMORY_REGISTERS abar = NULL;
 
-    StorPortDebugPrint(0, "AhciAdapterReset()\n");
+    DebugPrint("AhciAdapterReset()\n");
 
     abar = adapterExtension->ABAR_Address;
     if (abar == NULL) // basic sanity
+    {
         return FALSE;
+    }
 
     // HR -- Very first bit (lowest significant)
     ghc = AHCI_Global_HBA_CONTROL_HR;
     StorPortWriteRegisterUlong(adapterExtension, &abar->GHC, ghc);
 
-    for (ticks = 0; (ticks < 50) &&
-                    (StorPortReadRegisterUlong(adapterExtension, &abar->GHC) == 1);
-                    StorPortStallExecution(20000), ticks++);
+    for (ticks = 0; ticks < 50; ++ticks)
+    {
+        if ((StorPortReadRegisterUlong(adapterExtension, &abar->GHC) & AHCI_Global_HBA_CONTROL_HR) == 0)

```

Comment:
Do the read on one line into a temp, and compare the temp with the mask in the next line.